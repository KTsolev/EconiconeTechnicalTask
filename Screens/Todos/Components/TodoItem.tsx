import React, {FunctionComponent} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {TodoModel} from '../Model/TodoModel';
import CheckBox from '@react-native-community/checkbox';
import {useDispatch} from 'react-redux';
import {remove, complete} from '../TodoSlice';


export const TodoItem: FunctionComponent<TodoItemProps> = ({
  item,
  isReadOnly,
}): JSX.Element => {
  const dispatch = useDispatch();

  const deleteItem = () => dispatch(remove(item?.id));
  const completeItem = (value: any) =>
    dispatch(complete({id: item?.id, value}));

  return (
    <View>
      <CheckBox value={item?.completed} onValueChange={completeItem} />
      <Text>{item?.title}</Text>
      <TouchableOpacity onPress={deleteItem}>
        <Text>X</Text>
      </TouchableOpacity>
    </View>
  );
};

type TodoItemProps = {
  item?: TodoModel | null;
  isReadOnly: boolean; // the paragraph is optional
};
