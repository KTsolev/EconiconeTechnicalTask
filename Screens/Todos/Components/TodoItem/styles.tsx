import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
    marginVertical: 4,
    paddingVertical: 14,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
  text: {
    flex: 0.8,
    fontSize: 16,
    paddingLeft: 4,
    color: '#282C34',
  },
  completed: {
    textDecorationLine: 'line-through',
    textDecorationStyle: 'solid',
    textDecorationColor: '#000',
  },
  button: {
    width: 20,
    height: 20,
    padding: 14,
    borderRadius: 14,
    justifyContent: 'center',
    alignItems: 'center',
  },
  deleteButton: {
    backgroundColor: '#ff3344',
  },
  editButton: {
    backgroundColor: '#334488',
  },
});
