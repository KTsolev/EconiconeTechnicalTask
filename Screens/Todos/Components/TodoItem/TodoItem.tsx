import React, {FunctionComponent} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {TodoModel} from '../../Models/TodoModel';
import CheckBox from '@react-native-community/checkbox';
import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
import {faXmark} from '@fortawesome/free-solid-svg-icons/faXmark';
import {faPencil} from '@fortawesome/free-solid-svg-icons/faPencil';
import {useDispatch} from 'react-redux';
import {removeTodo, completeTodo} from '../../Todoslice/TodoSlice';
import {styles} from './styles';
import {AddNewItem} from '../AddNewTodo/AddNewTodo';

export const TodoItem: FunctionComponent<TodoItemProps> = ({
  item,
  index,
  isReadOnly,
  toggleEditTask,
}): JSX.Element | null => {
  const dispatch = useDispatch();

  const deleteItem = () => item && dispatch(removeTodo(item.id));
  const completeItem = (value: any) =>
    item && dispatch(completeTodo({id: item?.id, value}));

  if (!item) {
    return null;
  }

  if (!isReadOnly) {
    return (
      <View style={styles.container}>
        <CheckBox
          testID="complete"
          value={item.completed}
          onValueChange={completeItem}
        />
        <Text style={[styles.text, item.completed ? styles.completed : null]}>
          {item?.title}
        </Text>
        <TouchableOpacity
          testID="edit"
          style={[styles.button, styles.editButton]}
          onPress={() => toggleEditTask(index, !isReadOnly)}>
          <FontAwesomeIcon icon={faPencil} color="#fff" />
        </TouchableOpacity>
        <TouchableOpacity
          testID="delete"
          style={[styles.button, styles.deleteButton]}
          onPress={deleteItem}>
          <FontAwesomeIcon icon={faXmark} color="#fff" />
        </TouchableOpacity>
      </View>
    );
  } else {
    return (
      <View style={styles.container}>
        <AddNewItem
          item={item}
          closeEditTask={() => toggleEditTask(index, !isReadOnly)}
        />
      </View>
    );
  }
};

type TodoItemProps = {
  item?: TodoModel | null;
  isReadOnly: boolean;
  index: number;
  toggleEditTask: (index: number, value: boolean) => void;
};
