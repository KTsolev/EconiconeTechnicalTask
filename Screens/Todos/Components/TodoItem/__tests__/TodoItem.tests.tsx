import React from 'react';
import {TodoItem} from '../TodoItem';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import {todos} from '../../../../../__mocks__/todos.mock';

const mockDispatchFn = jest.fn();
const mockToggleEditTask = jest.fn();

jest.mock('uniqid', () => () => '1911caln7e0l28wbt2j');

jest.mock('react-redux', () => ({
  useSelector: (selector: () => any) => selector(),
  useDispatch: () => mockDispatchFn,
}));

describe('AddNewItem tests', () => {
  it('should render component', async () => {
    const tree = renderer.create(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={false}
        toggleEditTask={mockToggleEditTask}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should handle case when item is null', async () => {
    const tree = renderer.create(
      <TodoItem
        key={2}
        item={null}
        index={2}
        isReadOnly={false}
        toggleEditTask={mockToggleEditTask}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should handle complete task', async () => {
    const {getByTestId, toJSON} = render(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={false}
        toggleEditTask={mockToggleEditTask}
      />,
    );
    const checkBox = getByTestId('complete');
    fireEvent(checkBox, 'onValueChange', {nativeEvent: {}});

    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).toHaveBeenCalledWith({
      payload: {
        id: 'l28uy38o',
        value: {
          nativeEvent: {},
        },
      },
      type: 'todos/completeTodo',
    });
  });

  it('should handle delete task', async () => {
    const {getByTestId, toJSON} = render(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={false}
        toggleEditTask={mockToggleEditTask}
      />,
    );
    const button = getByTestId('delete');
    fireEvent.press(button);

    expect(toJSON()).toMatchSnapshot();
    expect(mockDispatchFn).toHaveBeenCalledWith({
      payload: 'l28uy38o',
      type: 'todos/removeTodo',
    });
  });

  it('should handle edit task', async () => {
    const {getByTestId, toJSON} = render(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={false}
        toggleEditTask={mockToggleEditTask}
      />,
    );
    const button = getByTestId('edit');
    fireEvent.press(button);

    expect(toJSON()).toMatchSnapshot();
    expect(mockToggleEditTask).toHaveBeenCalled();
  });

  it('should handle case readOnly task', async () => {
    const {toJSON} = render(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={true}
        toggleEditTask={mockToggleEditTask}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should handle call mockToggleEditTask task', async () => {
    const {getByTestId, toJSON} = render(
      <TodoItem
        key={2}
        item={todos[2]}
        index={2}
        isReadOnly={true}
        toggleEditTask={mockToggleEditTask}
      />,
    );

    const input = getByTestId('input');
    const button = getByTestId('createButton');

    fireEvent.changeText(input, 'sampletask');
    fireEvent.press(button);

    expect(toJSON()).toMatchSnapshot();
    expect(mockToggleEditTask).toHaveBeenCalled();
  });
});
