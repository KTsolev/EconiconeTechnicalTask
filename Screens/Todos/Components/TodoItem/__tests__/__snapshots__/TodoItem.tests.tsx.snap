// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewItem tests should handle call mockToggleEditTask task 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 4,
        "paddingHorizontal": 10,
        "paddingVertical": 14,
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": false,
        }
      }
      accessibilityValue={
        Object {
          "text": "off",
        }
      }
      accessible={true}
      pointerEvents="auto"
      testID="completeEdit"
    >
      <RNCCheckbox
        forwardedRef={null}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            undefined,
          ]
        }
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#888888",
          "borderRadius": 6,
          "borderWidth": 1,
          "flex": 0.82,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "position": "relative",
        }
      }
    >
      <TextInput
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Enter new task"
        style={
          Array [
            Object {
              "color": "#888888",
              "fontWeight": "600",
              "maxHeight": 120,
              "minHeight": 40,
              "padding": 10,
            },
          ]
        }
        testID="input"
        value="sampletask"
      />
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1c90ff",
          "borderRadius": 6,
          "flex": 0.5,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "maxHeight": 68,
          "maxWidth": 184,
          "minHeight": 40,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="createButton"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Edit Task
      </Text>
    </View>
  </View>
</View>
`;

exports[`AddNewItem tests should handle case readOnly task 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 4,
        "paddingHorizontal": 10,
        "paddingVertical": 14,
      }
    }
  >
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": false,
        }
      }
      accessibilityValue={
        Object {
          "text": "off",
        }
      }
      accessible={true}
      pointerEvents="auto"
      testID="completeEdit"
    >
      <RNCCheckbox
        forwardedRef={null}
        onValueChange={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            undefined,
          ]
        }
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#888888",
          "borderRadius": 6,
          "borderWidth": 1,
          "flex": 0.82,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "position": "relative",
        }
      }
    >
      <TextInput
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Enter new task"
        style={
          Array [
            Object {
              "color": "#888888",
              "fontWeight": "600",
              "maxHeight": 120,
              "minHeight": 40,
              "padding": 10,
            },
          ]
        }
        testID="input"
        value="Fix ability to togle a task"
      />
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1c90ff",
          "borderRadius": 6,
          "flex": 0.5,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "maxHeight": 68,
          "maxWidth": 184,
          "minHeight": 40,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="createButton"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Edit Task
      </Text>
    </View>
  </View>
</View>
`;

exports[`AddNewItem tests should handle case when item is null 1`] = `null`;

exports[`AddNewItem tests should handle complete task 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": true,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "on",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="complete"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#282C34",
          "flex": 0.8,
          "fontSize": 16,
          "paddingLeft": 4,
        },
        Object {
          "textDecorationColor": "#000",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    Fix ability to togle a task
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#334488",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="edit"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              61504,
              9999,
              "pencil-alt",
            ],
            "f303",
            "M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z",
          ],
          "iconName": "pencil",
          "prefix": "fas",
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff3344",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="delete"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [
              128473,
              10005,
              10006,
              10060,
              215,
              "close",
              "multiply",
              "remove",
              "times",
            ],
            "f00d",
            "M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z",
          ],
          "iconName": "xmark",
          "prefix": "fas",
        }
      }
    />
  </View>
</View>
`;

exports[`AddNewItem tests should handle delete task 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": true,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "on",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="complete"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#282C34",
          "flex": 0.8,
          "fontSize": 16,
          "paddingLeft": 4,
        },
        Object {
          "textDecorationColor": "#000",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    Fix ability to togle a task
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#334488",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="edit"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              61504,
              9999,
              "pencil-alt",
            ],
            "f303",
            "M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z",
          ],
          "iconName": "pencil",
          "prefix": "fas",
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff3344",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="delete"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [
              128473,
              10005,
              10006,
              10060,
              215,
              "close",
              "multiply",
              "remove",
              "times",
            ],
            "f00d",
            "M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z",
          ],
          "iconName": "xmark",
          "prefix": "fas",
        }
      }
    />
  </View>
</View>
`;

exports[`AddNewItem tests should handle edit task 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": true,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "on",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="complete"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#282C34",
          "flex": 0.8,
          "fontSize": 16,
          "paddingLeft": 4,
        },
        Object {
          "textDecorationColor": "#000",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    Fix ability to togle a task
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#334488",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="edit"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              61504,
              9999,
              "pencil-alt",
            ],
            "f303",
            "M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z",
          ],
          "iconName": "pencil",
          "prefix": "fas",
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff3344",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="delete"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [
              128473,
              10005,
              10006,
              10060,
              215,
              "close",
              "multiply",
              "remove",
              "times",
            ],
            "f00d",
            "M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z",
          ],
          "iconName": "xmark",
          "prefix": "fas",
        }
      }
    />
  </View>
</View>
`;

exports[`AddNewItem tests should render component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginVertical": 4,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": true,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "on",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="complete"
  >
    <RNCCheckbox
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          undefined,
        ]
      }
      value={true}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#282C34",
          "flex": 0.8,
          "fontSize": 16,
          "paddingLeft": 4,
        },
        Object {
          "textDecorationColor": "#000",
          "textDecorationLine": "line-through",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    Fix ability to togle a task
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#334488",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="edit"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              61504,
              9999,
              "pencil-alt",
            ],
            "f303",
            "M421.7 220.3L188.5 453.4L154.6 419.5L158.1 416H112C103.2 416 96 408.8 96 400V353.9L92.51 357.4C87.78 362.2 84.31 368 82.42 374.4L59.44 452.6L137.6 429.6C143.1 427.7 149.8 424.2 154.6 419.5L188.5 453.4C178.1 463.8 165.2 471.5 151.1 475.6L30.77 511C22.35 513.5 13.24 511.2 7.03 504.1C.8198 498.8-1.502 489.7 .976 481.2L36.37 360.9C40.53 346.8 48.16 333.9 58.57 323.5L291.7 90.34L421.7 220.3zM492.7 58.75C517.7 83.74 517.7 124.3 492.7 149.3L444.3 197.7L314.3 67.72L362.7 19.32C387.7-5.678 428.3-5.678 453.3 19.32L492.7 58.75z",
          ],
          "iconName": "pencil",
          "prefix": "fas",
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ff3344",
        "borderRadius": 14,
        "height": 20,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 14,
        "width": 20,
      }
    }
    testID="delete"
  >
    <i
      className="fa undefined"
      color="#fff"
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [
              128473,
              10005,
              10006,
              10060,
              215,
              "close",
              "multiply",
              "remove",
              "times",
            ],
            "f00d",
            "M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z",
          ],
          "iconName": "xmark",
          "prefix": "fas",
        }
      }
    />
  </View>
</View>
`;
